lib/workers/repository/process/lookup/index.ts

```diff
@@ -541,7 +541,14 @@ export async function lookupUpdates(
     // Add digests if necessary
     if (supportsDigests(config.datasource)) {
       if (config.currentDigest) {
-        if (!config.digestOneAndOnly || !res.updates.length) {
+        // Custom datasources should not run a digest update
+        // on the current version if it already runs a version update.
+        if (
+          !(
+            config.digestOneAndOnly ?? config.datasource.startsWith('custom.')
+          ) ||
+          !res.updates.length
+        ) {
```

Comment:
I'd like to convert this logic to an if/else to reduce the amount of negating. The `if` path can have logger.trace() inside it and the else then will contain the res.updates.push()

Reply:
(for future readability - the proposed change has definitely crossed a threshold for readability, maybe already had)

Reply:
I tried a few logic combinations and they are all very much unreadable, so I resorted to a boolean that will hopefully add some more to the semantics.

This is, unfortunately, still all very hacky, but I think a more sustainable solution will require deeper insights into the renovate code bases and use cases than I currently possess.